### DISCUSSION TITLE
AFix integration with QFormat

### DISCUSSION THREAD

**Discussion by mrberman87** [SpinalHDL/SpinalHDL#1396] [Ideas]
Would the ability to use the QFormat type to define AFix sizes be something worth while?

I am working with the AFix types for some filter design, and this seems like a logical extension to me, but I don't know how the rest of the community would want this?  This is something I could implement if there is desire.

**Comment by Dolu1990** [comment]
An api for QFormat definition in afix is present as : 

```scala
val miaou = AFix.UQ(10 bits, 8 bits)
```

Are you thinking about other things than "just" defining QFormat numbers via AFix ?

**Reply by mrberman87** [reply]
I was thinking of using the QFormat type from core/src/main/scala/spinal/core/QFormat.scala.  There is a slight difference, and is handled in the fixTo overload [here](https://github.com/SpinalHDL/SpinalHDL/blob/b34cdb378eb5a2061e92405a7ca1ed16d21f146a/core/src/main/scala/spinal/core/AFix.scala#L1056).  The AFix.UQ takes an integer width and fractional width, as opposed to the QFormat working with a total width and a fractional width.  This is simple arithmetic, but makes working between the two a little more cumbersome than it could be.

The extent of what I was thinking was an overload of the apply methods in the object AFix, and to add a rounding type to the fixTo method that takes a QFormat for the sizing.

```scala
def apply(Q: QFormat): AFix = AFix(Q.width-Q.fraction exp, -Q.fraction exp, Q.signed)

def fixTo(Q: QFormat, roundType: RoundType): AFix = {
  val res = AFix(Q)  // assuming the above apply method is added
  res := this.fixTo(res, roundType)
  res
}
```

**Reply by Dolu1990** [reply]
Ahh sure, this can be added :D
Feel free to PR
